client/web/src/site-admin/SiteAdminWebhookCreatePage.tsx

```diff
@@ -0,0 +1,303 @@
+            )
+            setKindsToUrls(kindToUrlMap)
+            // If there are no external services, then the warning is shown and webhook creation is blocked
+            if (kindToUrlMap.size > 0) {
+                const extSvcKinds = new Set(extSvcKindsOrError.nodes.map(node => node.kind))
+                const kindsArray = Array.from(extSvcKinds)
+                setKinds(kindsArray)
+                const currentKind = kindsArray[0]
+                // we always generate a secret once and assign it to the webhook. Bitbucket Cloud special case
+                // is handled is an Input and during GraphQL query creation.
+                setWebhook(webhook => ({
+                    ...webhook,
+                    secret: generateSecret(),
+                    codeHostURN: kindToUrlMap.get(currentKind)?.[0] || '',
+                    codeHostKind: currentKind,
+                }))
+            }
+            setLoading(false)
+        }
+    }, [extSvcKindsOrError])
+
+    const onCodeHostTypeChange = useCallback<React.ChangeEventHandler<HTMLSelectElement>>(
+        event => {
+            const selected = event.target.value as ExternalServiceKind
+            setWebhook(webhook => ({
+                ...webhook,
+                codeHostKind: selected,
+                codeHostURN: kindsToUrls.get(selected)?.[0] || '',
+            }))
+        },
+        [kindsToUrls]
+    )
+    const onCodeHostUrnChange = useCallback<React.ChangeEventHandler<HTMLSelectElement>>(event => {
+        setWebhook(webhook => ({ ...webhook, codeHostURN: event.target.value }))
+    }, [])
+    const onNameChange = useCallback((name: string): void => {
+        setWebhook(webhook => ({ ...webhook, name }))
+    }, [])
+    const onSecretChange = useCallback((secret: string): void => {
+        setWebhook(webhook => ({ ...webhook, secret: secret.length === 0 ? null : secret }))
+    }, [])
+
+    const [createWebhook, { error: createWebhookError, loading: creationLoading }] = useMutation<
+        CreateWebhookResult,
+        CreateWebhookVariables
+    >(CREATE_WEBHOOK_QUERY, { onCompleted: data => history.push(`/site-admin/webhooks/${data.createWebhook.id}`) })
+
+    return (
+        <Container>
+            <PageTitle title="Incoming webhook" />
+            <PageHeader
+                path={[{ icon: mdiCog }, { to: '/site-admin/webhooks', text: 'Incoming webhooks' }, { text: 'Create' }]}
+                className="mb-3"
+                headingElement="h2"
+            />
+
+            {isErrorLike(extSvcKindsOrError) && (
+                <Alert variant="danger" className="mt-2">
+                    Error during getting external services.
+                </Alert>
+            )}
+            {!isErrorLike(extSvcKindsOrError) && kindsToUrls.size === 0 && (
+                <Alert variant="warning" className="mt-2">
+                    Please add an external service to proceed with webhooks creation.
```

Comment:
```suggestion
                    Please add a code host connection to proceed with webhooks creation.
```

And also link to the page.